rpcs3/Emu/RSX/GL/GLGSRender.cpp

```diff
@@ -1898,12 +1898,6 @@ void GLGSRender::do_local_task(rsx::FIFO_state state)
 
 	rsx::thread::do_local_task(state);
 
-	if (state == rsx::FIFO_state::lock_wait)
```

Comment:
Why remove this? This is intended. In case of lock_wait only thing supposed to execute is possible texture cache flush in case a thread is stuck in access_violation_handler while RSX is waiting for semaphore acquire somewhere. It is not desired to actually do anything that takes more than that to avoid desync possibilities.

Reply:
To handle ::native_ui flag flip.

Reply:
Why can it not be deffered? It is meant to be deferrable (flip at earliest convenience). Its not meant to be dealt with immediately. For example, it should never even flip if the internal state is not consistent, e.g inside a begin_end state.

Reply:
Yea, but inside semaphore_acquire it might wait for the flip semapgore itself, may need a better fix though.

Reply:
Isn't it set by vblank thread? There should be no issues there afaict

Reply:
nope, vbalnk thread uses 0xFED subcommand at the rsx syscall, flip notification is 0xFEC.

Reply:
Right, in this case, native flip should not modify hw state at all, that is only allowed for system requested flip. Comment out this block of code instead of removing it, I'll rework the flip framework to take the semaphores into account.